staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer.go

```diff
@@ -116,6 +134,15 @@ func (f *dynamicSharedInformerFactory) WaitForCacheSync(stopCh <-chan struct{})
 	return res
 }
 
+func (f *dynamicSharedInformerFactory) Shutdown() {
+	f.lock.Lock()
+	f.shuttingDown = true
+	f.lock.Unlock()
+
+	// Will return immediately if there is nothing to wait for.
+	f.wg.Wait()
```

Comment:
```suggestion
	// Will return immediately if there is nothing to wait for.
	defer f.wg.Wait()
	
	f.lock.Lock()
	defer f.lock.Unlock()
	f.shuttingDown = true
```

(feels better to unlock via defer, however you want to do that)